joblib/test/test_memory.py

```diff
@@ -162,58 +162,52 @@ def name_collision(x):
 
     b = name_collision
 
-    with warnings.catch_warnings(record=True) as w:
-        # Cause all warnings to always be triggered.
-        warnings.simplefilter("always")
-        a(1)
-        b(1)
+    a(1)
+    b(1)
+    assert len(recwarn) == 1
+    w = recwarn.pop(JobLibCollisionWarning)
+    assert "collision" in str(w.message)
 
-        assert len(w) == 1
-        assert "collision" in str(w[-1].message)
 
-
-def test_memory_warning_lambda_collisions(tmpdir):
+def test_memory_warning_lambda_collisions(tmpdir, recwarn):
     # Check that multiple use of lambda will raise collisions
     memory = Memory(cachedir=tmpdir.strpath, verbose=0)
     # For isolation with other tests
-    memory.clear()
+    memory.clear(warn=False)
```

Comment:
This gives a `DeprecationWarning` which isn't ignored for Python >= 3. So the assertion below failed certain jobs on CI. I hope this is fine.